### PR TITLE
Add unit tests for BillableUsageStatusConsumer edge cases

### PR DESCRIPTION
**Pull request by vbusch**
Jira issue: SWATCH-5162

## Description
Added two new test cases to `BillableUsageStatusConsumerTest`:

  1. **`testWhenStatusIsNullThenRemittanceIsNotUpdated`** - Validates that status messages with null status values do not modify existing remittances
  2. **`testWhenStatusUpdateReferencesMultipleRemittancesThenAllAreUpdated`** - Validates that a single status message can update multiple remittance UUIDs in one batch operation

* No production code changes required - the existing implementation already correctly handles both scenarios

## Testing
  `mvn test -Dtest=BillableUsageStatusConsumerTest -pl swatch-billable-usage`

  Closes SWATCH-5162

## Summary by CodeRabbit

* **Tests**
  * Expanded coverage for remittance processing behavior in additional scenarios.
  * Added checks to ensure pending remittances stay unchanged when no status is provided.
  * Added validation that multiple related remittances are all updated correctly on success, including billed date and error status handling.

### CODE REVIEW

**Review by Mantzounis — DISMISSED** [review]
lgtm!

File: swatch-billable-usage/src/test/java/com/redhat/swatch/billable/usage/services/BillableUsageStatusConsumerTest.java [hunk 1]
```diff
@@ -181,13 +181,58 @@ void testWhenHandlingStatusThenUpdatedAtIsPopulated() {
...
+    var remittance1 = givenExistingRemittance();
+    var remittance2 = givenExistingRemittance();
+    var remittance3 = givenExistingRemittance();
+
+    var successMessage =
+        createBillableUsageAggregate(
+            Status.SUCCEEDED, null, BILLED_ON, remittance1, remittance2, remittance3);
+
+    whenSendResponse(successMessage);
+    Awaitility.await()
```
Comment (TommasoLencioni) [inline swatch-billable-usage/src/test/java/com/redhat/swatch/billable/usage/services/BillableUsageStatusConsumerTest.java]: Could `verifyUpdateForSuccess` be used here?
```suggestion
    Awaitility.await().untilAsserted(this::verifyUpdateForSuccess);
```

Comment (vbusch) [reply swatch-billable-usage/src/test/java/com/redhat/swatch/billable/usage/services/BillableUsageStatusConsumerTest.java]: Nice catch!  Updated.

**Review by TommasoLencioni — APPROVED** [review]
